libs/vm-modules/examples/boston_house_prices/main.cpp

```diff
@@ -0,0 +1,140 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "ml/dataloaders/ReadCSV.hpp"
+#include "vm/module.hpp"
+#include "vm_modules/core/print.hpp"
+#include "vm_modules/core/system.hpp"
+#include "vm_modules/math/tensor.hpp"
+#include "vm_modules/ml/ml.hpp"
+
+#include <cstdint>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+using System     = fetch::vm_modules::System;
+using DataType   = fetch::vm_modules::math::VMTensor::DataType;
+using TensorType = fetch::math::Tensor<DataType>;
+
+// read the weights and bias csv files
+fetch::vm::Ptr<fetch::vm_modules::math::VMTensor> read_csv(
+    fetch::vm::VM *vm, fetch::vm::Ptr<fetch::vm::String> const &filename, bool transpose)
+{
+  TensorType tensor = fetch::ml::dataloaders::ReadCSV<TensorType>(filename->str, 0, 0, transpose);
+  return vm->CreateNewObject<fetch::vm_modules::math::VMTensor>(tensor);
+}
+
+fetch::vm::Ptr<fetch::vm_modules::math::VMTensor> read_csv_no_transpose(
+    fetch::vm::VM *vm, fetch::vm::Ptr<fetch::vm::String> const &filename)
+{
+  return read_csv(vm, filename, false);
+}
+
+int main(int argc, char **argv)
+{
+  // parse the command line parameters
+  System::Parse(argc, argv);
+
+  fetch::commandline::ParamsParser const &pp = System::GetParamsParser();
+
+  // ensure the program has the correct number of args
+  if (2u != pp.arg_size())
+  {
+    std::cerr << "Usage: " << pp.GetArg(0) << " [options] <filename> -- [script args]..."
+              << std::endl;
+    return 1;
+  }
+
+  // Reading file
+  std::ifstream file(pp.GetArg(1), std::ios::binary);
+  if (file.fail())
+  {
+    throw std::runtime_error("Cannot open file " + std::string(pp.GetArg(1)));
+  }
+  std::ostringstream ss;
+  ss << file.rdbuf();
+  const std::string source = ss.str();
+  file.close();
+
+  auto module = std::make_shared<fetch::vm::Module>();
+
+  fetch::vm_modules::System::Bind(*module);
+
+  fetch::vm_modules::ml::BindML(*module);
+
+  fetch::vm_modules::CreatePrint(*module);
+
+  module->CreateFreeFunction("read_csv", &read_csv);
```

Comment:
Could we rename to `readCSV`? This would follow the style of the other functions in the library.